### ISSUE TITLE
Typescript error or breaking change after minor upgrade from 57.0.3 to 57.0.7

### ISSUE THREAD

**Issue by doberkofler** [gajus/eslint-plugin-jsdoc#1509]
After upgrading from 57.0.3 to 57.0.7 my eslint configuration:
```ts
	{
		files: ['**/*.{ts,tsx}'],
		plugins: {jsdoc},
		settings: {
			jsdoc: {
				mode: 'typescript',
			},
		},
		rules: {
			...jsdoc.configs['flat/contents-typescript'].rules,
			...jsdoc.configs['flat/logical-typescript'].rules,
			...jsdoc.configs['flat/requirements-typescript'].rules,
			'jsdoc/check-property-names': 'off', // TODO: should be enabled
			'jsdoc/check-tag-names': 'off', // TODO: should be enabled
			'jsdoc/text-escaping': ['warn', {escapeHTML: true, escapeMarkdown: true}],
			'jsdoc/informative-docs': 'off', // TODO: should be enabled
			'jsdoc/lines-before-block': 'off', // TODO: should be enabled
			'jsdoc/match-description': 'off', // TODO: should be enabled
			'jsdoc/no-blank-block-descriptions': 'off', // TODO: should be enabled
			'jsdoc/require-jsdoc': 'off',
			'jsdoc/require-example': 'off',
			'jsdoc/require-returns': 'off', // TODO: should be enabled
			'jsdoc/tag-lines': 'off',
		},
	},
```

the following type errors are reported:

```bash
scripts/eslint_config.ts:132:49 - error TS2339: Property 'rules' does not exist on type 'Config<RulesRecord> | Config<RulesRecord>[] | LegacyConfig<RulesRecord, RulesRecord>'.
  Property 'rules' does not exist on type 'Config<RulesRecord>[]'.

132    ...jsdoc.configs['flat/contents-typescript'].rules,

scripts/eslint_config.ts:133:48 - error TS2339: Property 'rules' does not exist on type 'Config<RulesRecord> | Config<RulesRecord>[] | LegacyConfig<RulesRecord, RulesRecord>'.
  Property 'rules' does not exist on type 'Config<RulesRecord>[]'.

133    ...jsdoc.configs['flat/logical-typescript'].rules,

scripts/eslint_config.ts:134:53 - error TS2339: Property 'rules' does not exist on type 'Config<RulesRecord> | Config<RulesRecord>[] | LegacyConfig<RulesRecord, RulesRecord>'.
  Property 'rules' does not exist on type 'Config<RulesRecord>[]'.

134    ...jsdoc.configs['flat/requirements-typescript'].rules,
```

Reverting to 57.0.3 solves the problem.

## Environment

- Node version: 24.8.0
- ESLint version 9.35.0
- `eslint-plugin-jsdoc` version: 57.0.7
